File: java/two-sum/src/main/java/TwoSum.java
import java.util.HashMap;
import java.util.Map;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> possibleValues = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int currentValue = nums[i];
            if (possibleValues.containsKey(currentValue)) {
                return new int[]{possibleValues.get(currentValue), i};
            } else {
                possibleValues.put(target - currentValue, i);
            }
        }
        throw new IllegalArgumentException("No two sum solution");

    }

}

File: go/two-sum/two_sum.go
package two_sum

func twoSum(nums []int, target int) []int {
	possibleValues := make(map[int]int)
	for i, currentValue := range nums {
		if val, ok := possibleValues[target-currentValue]; ok {
			return []int{i, val}
		} else {
			possibleValues[currentValue] = i
		}
	}
	return []int{0, 0}
}
